### PR TITLE
change primary key migration logic

### PR DESCRIPTION
**Pull request by Faryshta**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | yes |
| Fixed issues | #11638 |

The bc break is that the `Schema::TYPE_PK` and similar constants were eliminated.

### CODE REVIEW

**Comment by Faryshta** [comment]
it contains a merge from the 2.0 branch at the very last commit to use the new tests. i can undo that last commit if this is too hard to review.

File: docs/guide/input-validation.md [hunk 1]
```diff
@@ -96,6 +96,27 @@ According to the above validation steps, an attribute will be validated if and o
...
+> public function rules()
+> {
+>     return [
+>         // ...
+>         'password' => [['password'], 'string', 'max' => 60],
+>     ];
+> }
+> ```
+>
+> TYou can use it in a child model:
```
Comment (lynicidn) [inline docs/guide/input-validation.md]: -T

**Comment by samdark** [comment]
Yes. Please do. I've merged master into 2.1.

**Comment by Faryshta** [comment]
@samdark done

**Comment by samdark** [comment]
What's the deal with all this formatting changes?

**Comment by Faryshta** [comment]
On the UPGRADE.md file? I will fix it, I had an 'markdown fixer' on my IDE but it seems too intrusive.

I made changes on some PHPDoc too but only about the line break. Shall I undo those too?

**Comment by samdark** [comment]
Well, it's hard to review :(

**Comment by Faryshta** [comment]
ok will simplify it tomorrow

**Comment by SilverFire** [comment]
The PR seems to be stuck.

@Faryshta, @samdark

**Pull request closed without merging**